### PR TITLE
Fix build settings resolution for external developers

### PR DESCRIPTION
**Pull request by ejensen**
**Description**:

[The README recommends](https://github.com/duckduckgo/iOS#developer-details) using an `ExternalDeveloper.xcconfig` file in order to supply values that differ from the `DuckDuckGoDeveloper.xcconfig`. The problem is that the project's targets have build setting customizations that override the `ExternalDeveloper.xcconfig` values. This PR removes target build settings that override developer configuration files. Allowing external developers to run the app without needing to modifying the project file.

**Steps to test this PR**:
1. Follow the steps in the ["Developer details" section of the README](https://github.com/duckduckgo/iOS#developer-details)
2. Open Xcode and verify the build settings for the target reflect the values assigned in `ExternalDeveloper.xcconfig` (particularly `DEVELOPMENT_TEAM`)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

**Comment by samsymons** [comment]
@ejensen Thanks so much for taking this on! 

I ran into some trouble getting this running with a DuckDuckGo developer account. Specifically, removing the manual `CODE_SIGN_STYLE` setting forces Xcode to try and create a managed provisioning profile, which fails since it needs a special entitlement:

```
Provisioning profile "iOS Team Provisioning Profile: com.duckduckgo.mobile.ios" doesn't include the com.apple.developer.web-browser entitlement.
```

Would it be possible to move the deleted settings into the `DuckDuckGoDeveloper` xcconfig file? That way it should work as expected for DDG devs, while also overriding settings as expected with an external developer config file. 🙂

**Comment by brindy** [comment]
Hi @ejensen - it's been a while since this was opened and there was no response to @samsymons.  Is this something you're planning to update?  Thanks!

**Comment by ejensen** [comment]
@samsymons @brindy commit 1f49941 adds the manually specified provisioning profile to the Duck Duck Go developer settings configuration

**Comment by brindy** [comment]
Thanks @ejensen! Sorry for the delay in replying.  @samsymons will hopefully get to this soon.

File: Configuration/DuckDuckGoDeveloper.xcconfig [hunk 1]
```diff
@@ -21,3 +21,9 @@ APP_ID = com.duckduckgo.mobile.ios
 
 // A prefix for group ids. Must start with "group.".
 GROUP_ID_PREFIX = group.com.duckduckgo
+
+// The code signing style needs to be specified manually because the DDG team has multiple provisioning profiles that match the team and bundle identifier
+CODE_SIGN_STYLE = Manual
+
+// The manualy specified provisioning profile
+PROVISIONING_PROFILE_SPECIFIER = "iOS 14 - Dev - App"
```
Comment (samsymons) [inline Configuration/DuckDuckGoDeveloper.xcconfig]: One quick request here: this'll include the quotes themselves in the provisioning profile specifier setting, after taking these out then this should work great!

```suggestion
PROVISIONING_PROFILE_SPECIFIER = iOS 14 - Dev - App
```

**Review by samsymons — CHANGES_REQUESTED** [review]

**Review by samsymons — APPROVED** [review]
LGTM! Thanks so much for this @ejensen! 👏

**Pull request merged**